cmd/syft/cli/convert/convert.go

```diff
@@ -0,0 +1,134 @@
+package convert
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/anchore/syft/internal/config"
+	"github.com/anchore/syft/internal/formats/cyclonedxjson"
+	"github.com/anchore/syft/internal/formats/cyclonedxxml"
+	"github.com/anchore/syft/internal/formats/spdx22json"
+	"github.com/anchore/syft/internal/formats/spdx22tagvalue"
+	"github.com/anchore/syft/internal/formats/syftjson"
+	"github.com/anchore/syft/internal/formats/table"
+	"github.com/anchore/syft/internal/log"
+	"github.com/anchore/syft/syft"
+	"github.com/anchore/syft/syft/sbom"
+	"github.com/hashicorp/go-multierror"
+)
+
+var ConvertibleFormats = []sbom.FormatID{
+	syftjson.ID,
+	spdx22json.ID,
+	spdx22tagvalue.ID,
+	cyclonedxjson.ID,
+	cyclonedxxml.ID,
+	table.ID,
+}
+
+func Run(ctx context.Context, app *config.Application, args []string) error {
+	log.Warn("convert is an experimental feature, run `syft convert -h` for help")
+	writer, err := makeWriter(app.Outputs, app.File)
+	if err != nil {
+		return err
+	}
+
+	defer func() {
+		if err := writer.Close(); err != nil {
+			log.Warnf("unable to write to report destination: %w", err)
+		}
+	}()
+
+	// this can only be a SBOM file
+	userInput := args[0]
+	f, err := os.Open(userInput)
+	if err != nil {
+		return fmt.Errorf("failed to open SBOM file: %w", err)
+	}
+
+	sbom, inputFormat, err := syft.Decode(f)
+	if err != nil {
+		return fmt.Errorf("failed to decode SBOM: %w", err)
+	}
+	f.Close()
+
+	if !isSupportedFormat(inputFormat.ID()) {
+		return fmt.Errorf("cannot convert from %s format", inputFormat.ID())
+	}
+
+	return writer.Write(*sbom)
+}
+
+func isSupportedFormat(format sbom.FormatID) bool {
+	for _, f := range ConvertibleFormats {
+		if format == f {
+			return true
+		}
+	}
+
+	return false
+}
+
+// makeWriter creates a sbom.Writer for output or returns an error. this will either return a valid writer
+// or an error but neither both and if there is no error, sbom.Writer.Close() should be called
+func makeWriter(outputs []string, defaultFile string) (sbom.Writer, error) {
+	outputOptions, formats, err := parseOptions(outputs, defaultFile)
+	if err != nil {
+		return nil, err
+	}
+
+	for _, f := range formats {
+		if !isSupportedFormat(f.ID()) {
+			return nil, fmt.Errorf("cannot convert to %s", f.ID())
+		}
+	}
+
+	writer, err := sbom.NewWriter(outputOptions...)
+	if err != nil {
+		return nil, err
+	}
+
+	for _, o := range outputOptions {
+		log.Debugf("writer options: %+v", o.Format)
+	}
+
+	return writer, nil
+}
+
+// parseOptions utility to parse command-line option strings and retain the existing behavior of default format and file
+func parseOptions(outputs []string, defaultFile string) (out []sbom.WriterOption, formats []sbom.Format, errs error) {
```

Comment:
Awesome! Makes sense why they are a little different - is there a way we can update it so that we don't copy and paste over with those small changes? Maybe allow writer to accept which are valid/supported formts?

cc @anchore/tools for additional comments if I'm totally off on this suggestion.

Reply:
Good point, I organized it the function can be used by both package and convert cmds.